### PR TITLE
[ENG-4888] Added the initial route, controller and template for the submit route

### PR DESCRIPTION
**Pull request by bp-cos**
-   Ticket: [ENG-4888]
-   Feature flag: n/a

## Purpose

Add the initial submit route and files

## Summary of Changes

Added the initial route, controller and template for the submit route

## Screenshot(s)
Screenshot 2024-02-27 at 11 31 28 AM

## Side Effects

It's rather vanilla since a few items are just placeholders

## QA Notes

[ENG-4888]: https://openscience.atlassian.net/browse/ENG-4888?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: app/preprints/submit/route.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    @service store!: Store;
+    @service theme!: Theme;
+    @service router!: RouterService;
+    @service metaTags!: MetaTags;
+    headTags?: HeadTagDef[];
+
+    buildRouteInfoMetadata() {
+        return {
+            osfMetrics: {
+                isSearch: true,
```
Comment (futa-ikeda) [inline app/preprints/submit/route.ts:21]:
```suggestion
```

File: app/preprints/submit/route.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+        try {
+            const provider = await this.store.findRecord('preprint-provider', args.provider_id);
+            this.theme.providerType = 'preprint';
+            this.theme.id = args.provider_id;
+            return {
+                provider,
+            };
+        } catch (e) {
+
+            this.router.transitionTo('not-found', `preprints/${args.provider_id}/discover`);
```
Comment (futa-ikeda) [inline app/preprints/submit/route.ts:37]:
```suggestion
            this.router.transitionTo('not-found', `preprints/${args.provider_id}/submit`);
```

**Review by futa-ikeda — CHANGES_REQUESTED** [review]

**Review by cslzchen — COMMENTED** [review]
Looks good and will approve after the two changes from Futa.

File: app/preprints/submit/controller.ts [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+import { inject as service } from '@ember/service';
+import Media from 'ember-responsive';
+
+export default class PreprintSubmitController extends Controller {
+    @service media!: Media;
+
+    get isMobile() {
+        return this.media.isMobile;
+    }
+}
```
Comment (cslzchen) [inline app/preprints/submit/controller.ts:12]: As @futa-ikeda mentioned, `is-mobile` is a builtin from `ember-responsive` now, so we don't need to wait for [his PR](https://github.com/CenterForOpenScience/ember-osf-web/pull/2153) to use `is-mobile` and remove this helper. We can do it here or use another PR to clean the code.

**Review by cslzchen — APPROVED** [review]
LGTM 🎆

**Review by futa-ikeda — APPROVED** [review]

**Pull request merged**